util/conversion/conversion.go

```diff
@@ -90,6 +93,44 @@ func UpdateReferenceAPIContract(ctx context.Context, c client.Client, ref *corev
 	return nil
 }
 
+// UpgradeReferenceAPIContract takes a client and object reference, queries the API Server for
+// the Custom Resource Definition and looks which one is the stored version available.
+//
+// The object passed as input is modified in place if an updated compatible version is found.
+// NOTE: In case CRDs are named incorrectly, this func is using an APIReader to list CRDs
+// to avoid to implicitly create an informer for CRDs which would leads to high memory consumption.
```

Comment:
NOTE: We will have to do this deprecation cycle once more when we drop the apiReader.

Reply:
I think we can do breaking changes in go code between minors, so we can drop previous signature

Reply:
> I think we can do breaking changes in go code between minors, so we can drop previous signature

Do you mean changing it directly now or dropping the old one with v1.2.x?

**Note**: The signature we eventually need once we drop the fallback is this one: `func UpdateReferenceAPIContract(ctx context.Context, c client.Client, ref *corev1.ObjectReference) error`. So if we have a way to do a breaking change to this func directly without cycling through other func names I would prefer not deprecating it at all.

Reply:
@vincepri  ^^

Reply:
I'm fine doing a breaking change here, given the implications of caching entire CRDs

Reply:
Okay then let's keep it simple regarding deprecation and rotating through func names. I'll do a breaking change now to add the apiReader and we'll do another one once we don't support the fallback for the wrong CRD names anymore.

Reply:
Done

Reply:
/retitle ⚠️  Update references with an APIReader as fallback